<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>NewDialog</class>
 <widget class="QDialog" name="NewDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>558</width>
    <height>323</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Dialog</string>
  </property>
  <property name="modal">
   <bool>true</bool>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <item>
    <widget class="QTabWidget" name="tabWidget">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="tab">
      <attribute name="title">
       <string>Basic</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_5">
       <item>
        <widget class="QGroupBox" name="preset_box">
         <property name="title">
          <string>Preset</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_6">
          <item>
           <widget class="QRadioButton" name="preset_no_pipeline">
            <property name="text">
             <string>No pipeline no cache</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QRadioButton" name="preset_no_pipeline_cache">
            <property name="text">
             <string>No pipeline with cache</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QRadioButton" name="preset_pipelined_bare">
            <property name="text">
             <string>Pipelined without hazard unit and without cache</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QRadioButton" name="preset_pipelined">
            <property name="text">
             <string>Pipelined with hazard unit and cache</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QRadioButton" name="preset_custom">
            <property name="text">
             <string>Custom</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="reset_at_compile">
         <property name="text">
          <string>Reset at compile time (reload after make)</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer_3">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="Line" name="line_2">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <item>
          <widget class="QLabel" name="label">
           <property name="text">
            <string>Elf executable: </string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="elf_file"/>
         </item>
         <item>
          <widget class="QPushButton" name="pushButton_browse">
           <property name="text">
            <string>Browse</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_core">
      <attribute name="title">
       <string>Core</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <property name="sizeConstraint">
        <enum>QLayout::SetDefaultConstraint</enum>
       </property>
       <item>
        <widget class="QCheckBox" name="pipelined">
         <property name="text">
          <string>Pipelined</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="delay_slot">
         <property name="text">
          <string>Delay slot</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="hazard_unit">
         <property name="title">
          <string>Hazard unit</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
         <property name="checked">
          <bool>false</bool>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <item>
           <widget class="QRadioButton" name="hazard_stall">
            <property name="text">
             <string>Stall when hazard is detected</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QRadioButton" name="hazard_stall_forward">
            <property name="text">
             <string>Stall or forward when hazard is detected</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_memory">
      <attribute name="title">
       <string>Memory</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_4">
       <item>
        <widget class="QCheckBox" name="mem_protec_write">
         <property name="text">
          <string>Program memory write protection</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="mem_protec_exec">
         <property name="text">
          <string>Data memory executable protection</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox">
         <property name="title">
          <string>Access time (in cycles)</string>
         </property>
         <layout class="QFormLayout" name="formLayout">
          <item row="0" column="0">
           <widget class="QLabel" name="label_read">
            <property name="text">
             <string>Read:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QSpinBox" name="mem_time_read">
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>999999999</number>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_write">
            <property name="text">
             <string>Write:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QSpinBox" name="mem_time_write">
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>999999999</number>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_burts">
            <property name="text">
             <string>Burst:</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QSpinBox" name="mem_time_burst">
            <property name="minimum">
             <number>0</number>
            </property>
            <property name="maximum">
             <number>999999999</number>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer_2">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_cache_program">
      <attribute name="title">
       <string>Program cache</string>
      </attribute>
     </widget>
     <widget class="QWidget" name="tab_cache_data">
      <attribute name="title">
       <string>Data cache</string>
      </attribute>
     </widget>
     <widget class="QWidget" name="tab_os_emulation">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <attribute name="title">
       <string>OS Emulation</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_7">
       <item>
        <widget class="QCheckBox" name="osemu_enable">
         <property name="text">
          <string>Enable emulation of operating system services</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="osemu_known_syscall_stop">
         <property name="text">
          <string>Stop on known system call</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="osemu_unknown_syscall_stop">
         <property name="text">
          <string>Stop on unknown system call</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="osemu_interrupt_stop">
         <property name="text">
          <string>Stop on interrupt entry</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="osemu_exception_stop">
         <property name="text">
          <string>Stop and step over exceptions (overflow, etc.)</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="QLabel" name="label_fs_root">
           <property name="text">
            <string>Filesystem root:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="osemu_fs_root"/>
         </item>
         <item>
          <widget class="QPushButton" name="osemu_fs_root_browse">
           <property name="text">
            <string>Browse</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <spacer name="verticalSpacer_4">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>21</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <property name="leftMargin">
      <number>9</number>
     </property>
     <property name="topMargin">
      <number>0</number>
     </property>
     <property name="rightMargin">
      <number>9</number>
     </property>
     <property name="bottomMargin">
      <number>0</number>
     </property>
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="pushButton_start_empty">
       <property name="text">
        <string>Start empty</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="pushButton_load">
       <property name="text">
        <string>Load machine</string>
       </property>
       <property name="default">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="pushButton_cancel">
       <property name="text">
        <string>Cancel</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
